patch/cisco-hwmon-pmbus_core-pec-support-check.patch

```diff
@@ -0,0 +1,39 @@
+From ca2339d26b3473aa4f85006c5caff8eb01fd5435 Mon Sep 17 00:00:00 2001
+From: Madhava Reddy Siddareddygari <msiddare@cisco.com>
+Date: Wed, 26 May 2021 11:25:14 -0700
+Subject: [PATCH] hwmon: (pmbus_core) Do not enable PEC if adapter doesn't
+ support it
+
+Added code to check if i2c bus has capability to support PEC [ Packet Error Correction ] and enable it only if both controller and bus support it
+
+I2C_FUNC_SMBUS_PEC can be used to check PEC capability
+
+Signed-off-by: Madhava Reddy Siddareddygari <msiddare@cisco.com>
+---
+ drivers/hwmon/pmbus/pmbus_core.c | 10 ++++++----
+ 1 file changed, 6 insertions(+), 4 deletions(-)
+
+diff --git a/drivers/hwmon/pmbus/pmbus_core.c b/drivers/hwmon/pmbus/pmbus_core.c
+index 1d7f950e6..16b20d48d 100644
+--- a/drivers/hwmon/pmbus/pmbus_core.c
++++ b/drivers/hwmon/pmbus/pmbus_core.c
+@@ -2044,10 +2044,12 @@ static int pmbus_init_common(struct i2c_client *client, struct pmbus_data *data,
+ 		data->has_status_word = true;
+ 	}
+ 
+-	/* Enable PEC if the controller supports it */
+-	ret = i2c_smbus_read_byte_data(client, PMBUS_CAPABILITY);
+-	if (ret >= 0 && (ret & PB_CAPABILITY_ERROR_CHECK))
+-		client->flags |= I2C_CLIENT_PEC;
++	/* Enable PEC if the controller and bus supports it */
```

Comment:
I’d re-order *controller* and *bus* in the comment, so it matches the order of the checks.

Reply:
Re-ordered code as per the comments